### DISCUSSION TITLE
4 controllers support on N64

### DISCUSSION THREAD

**Discussion by raed143** [darthcloud/BlueRetro#107] [Q&A]
Anyone managed to get 4 controllers working on N64. I get the first 2 working perfectly but controllers 3 and 4 keeps disconnecting, buttons do not always respond, serious lag. am using firmware v1.1.1 on an espressif ESP32 board.

**Comment by darthcloud** [comment]
I do but with a mix of PS4, PS5 and Xbox controller. Not every controller mix work well with high controller count.

What are the controller you are using?

**Reply by raed143** [reply]
I used 1 Nintendo 64 NSO controller, 3 Switch pro controllers. So if I used all same controllers it should work fine?

**Reply by darthcloud** [reply]
I theory using all SW ctrl should be fine but they do sometimes disconnect unlike PS and xbox. Something I need to look into.

Are your sw pro all Nintendo branded?

**Comment by raed143** [comment]
Yes all nintendo. But am mixing switch pro with n64 NSO. Didn't try all
switch pro.

**Reply by darthcloud** [reply]
Dont worry its the same thing just different plastic ;)
